models/DataObject/Data/Link.php

```diff
@@ -124,119 +139,156 @@ public function setLinktype(?string $linktype): static
         return $this;
     }
 
-    public function getTarget(): string
+    public function getTarget(): ?string
     {
         return $this->target;
     }
 
-    public function setTarget(string $target): static
+    /**
+     * @return $this
+     */
+    public function setTarget(?string $target): static
```

Comment:
Making the parameter nullable is a breaking change, right? (see: https://3v4l.org/TLtr7)

Reply:
But it was nullable before. Its a bug of Pimcore 11. The serialized data in the db have null in the attributes. So whats your idea to fix this?

Reply:
If it's a bug, we should fix it, even if it's breaking. But we should probably add a changelog entry...

Reply:
what about replacing not set values in the serialized string before it deserialized here 
https://github.com/pimcore/pimcore/blob/67b064e83c439b72d9d90bbff22ebc82cd56afc2/models/DataObject/ClassDefinition/Data/Link.php#L94 
🤔 not cleanest solution but thinking something to avoid the bc break
put all the non-nullable properties in an array, loop each eg. `text` regex to find 
`:8:{s:7:" * text";s:15:" * `
and replace with
`:8:{s:7:" * text";s:0:"";s:15:" * ` or int

Reply:
nevermind on changing the serialized data, if it's a bug, it would be better bc-breaking than be broken.
The chance that some project would extend this Link object might be low.

So before merging, we would need a line in the upgrade notes like as suggested by Jacob

Reply:
@kingjia90 Done https://github.com/pimcore/pimcore/pull/15577/commits/2cfecdacb6705e755d2ff6be30383c8a4ac1a3a6